modules/blocks/chain.js

```diff
@@ -704,7 +704,7 @@ __private.popLastBlock = function(oldLastBlock, cb) {
 						oldLastBlock.transactions.reverse(),
 						transaction =>
 							__private
-								.undoStep(transaction, secondLastBlock, tx)
+								.undoStep(transaction, oldLastBlock, tx)
```

Comment:
If changes can be credited to wrong round, then its a critical thing. How does this get through from round tests? should not we include some test cases for it?

Reply:
@nazarhussain See description of this PR, I added some tests in https://github.com/LiskHQ/lisk/pull/2035, I can add more when I'll be refactoring functional system rounds tests.